### ISSUE TITLE
docker-env doesn't behave as expected when ',' in the string

### ISSUE THREAD

**Issue by praveenkumar** [minishift/minishift#345]
If we put comma in the string for a docker-env then it break it to list and not take as string but this is not case for docker-machine where `engine-env` argument take it as expected.

Note: I put `fmt.Println` in the source to check those details.

```
$ ./minikube start --docker-env HTTP_PROXY=http://fedora:fedora123@10.70.49.109:3128 --docker-env NO_PROXY="*xip.io,localhost,172.11.2.1" --iso-url=file:/Users/prkumar/minishift-centos7.iso --show-libmachine-logs --cpus 4 --memory 4096
Start have something [HTTP_PROXY=http://fedora:fedora123@10.70.49.109:3128 NO_PROXY=*xip.io localhost 172.11.2.1]


$ docker-machine create centos --driver virtualbox --engine-env "NO_PROXY=*.xip.io,localhost,172.10.1.1" --virtualbox-boot2docker-url=file:/Users/prkumar/minishift-centos7.iso
Copying certs to the local machine directory...
Copying certs to the remote machine...
Something messy {[] []  [NO_PROXY=*.xip.io,localhost,172.10.1.1] false [] [provider=virtualbox]  devicemapper false true [] https://get.docker.com}
```

**Comment by praveenkumar** [comment]
@jimmidyson Do you know if that options worked before for multiple value for NO_PROXY as docker-env?

**Comment by jimmidyson** [comment]
No idea sorry.

**Comment by hferentschik** [comment]
@praveenkumar Could it be because we are using StringSliceVar to pass the flag value allowing multiple values for this flag. The default separator might be the comma.

**Comment by hferentschik** [comment]
On second thought, this is probably about being able to specify the flag in general multiple times. Still, have you verified what comes out of this. Also, you should be able to step through the program to see where the unintended split happens.

**Comment by praveenkumar** [comment]
@hferentschik After looking it to kubernetes side I found that StringSliceVar shouldn't use for this purpose and stringArrayVar what we needed refered https://github.com/spf13/viper/issues/112 will sent a PR soon and then move ahead with proxy work.

**Comment by hferentschik** [comment]
> After looking it to kubernetes side I found that StringSliceVar shouldn't use for this purpose and stringArrayVar what we needed refered

+1 :-)

**Comment by hferentschik** [comment]
Merged via pull request #350
